### PR TITLE
chore: update prometheus dependency for service discovery

### PR DESCRIPTION
**Pull request by lesam**
Closes: https://github.com/influxdata/kapacitor/issues/2532

There were significant reformatting changes in the prometheus service discovery
packages and their configuration, but it seems like the logic is mostly similar.

###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated

### CODE REVIEW

**Comment by lesam** [comment]
Do we want a changelog for this?

File: Dockerfile_build_ubuntu32 [hunk 1]
```diff
@@ -1,58 +0,0 @@
-FROM 32bit/ubuntu:16.04
```
Comment (lesam) [inline Dockerfile_build_ubuntu32:1]: Delete the 32 bit tests since we will soon be deleting the 32 bit build.

File: go.mod [hunk 2]
```diff
@@ -3,112 +3,75 @@ module github.com/influxdata/kapacitor
...
 	github.com/onsi/ginkgo v1.14.2 // indirect
 	github.com/onsi/gomega v1.10.3 // indirect
-	github.com/pborman/uuid v0.0.0-20160209185913-a97ce2ca70fa // indirect
-	github.com/pkg/errors v0.8.1
-	github.com/prometheus/common v0.9.1
-	github.com/prometheus/prometheus v0.0.0-00010101000000-000000000000
+	github.com/pierrec/lz4 v2.5.2+incompatible // indirect
+	github.com/pkg/errors v0.9.1
+	github.com/prometheus/common v0.20.0
+	github.com/prometheus/prometheus v1.8.2-0.20210331101223-3cafc58827d1
```
Comment (lesam) [inline go.mod:52]: This is actually tag v2.26.0 of prometheus, but https://github.com/prometheus/prometheus/issues/6048 prevents us from using the semver name, we have to specify the commit instead.

File: services/scraper/service.go [hunk 3]
```diff
@@ -42,13 +49,23 @@ type Service struct {
 
 	discoverers []Discoverer
 
-	// TargetManager represents a scraping/discovery manager
-	mgr interface {
-		ApplyConfig(cfg *config.Config) error
-		Stop()
-		Start()
```
Comment (lesam) [inline services/scraper/service.go:49]: The prometheus fork we had was just to split `Run` into `Start` and `Wait`. But the latest version of prometheus looks like it should support calling `Stop` before `Run` in racy cases, so I don't think we need the fork any longer.

**Comment by lesam** [comment]
Tested: https://gist.github.com/lesam/1120b58a5261a2b7eee8270ce9426727

File: services/scraper/service.go (hunk 3 above)
Comment (docmerlin) [reply services/scraper/service.go:49]: awesome!

**Review by docmerlin — APPROVED** [review]

File: services/marathon/config.go [hunk 4]
```diff
@@ -27,11 +28,11 @@ type Config struct {
 	SSLServerName string `toml:"ssl-server-name" override:"ssl-server-name"`
 	// Use SSL but skip chain & host verification
 	InsecureSkipVerify bool `toml:"insecure-skip-verify" override:"insecure-skip-verify"`
+	// This previously supported Timeout but it was removed, see https://github.com/prometheus/common/pull/123
```
Comment (lesam) [inline services/marathon/config.go:31]: This needs a docs change.

**Pull request merged**